src/middlewares.rs

```diff
@@ -18,30 +21,66 @@ pub struct UserState {
 
 pub type CookieUserState = Extension<UserState>;
 
+impl<'a> From<UserState> for Cookie<'a> {
+    fn from(userstate: UserState) -> Self {
+        let userstate_json =
+            serde_json::to_string(&userstate).expect("failed to serialize to json");
+
+        let userstate_base64 = STANDARD_NO_PAD.encode(userstate_json);
+
+        Cookie::build(("state", userstate_base64))
+            .http_only(true)
+            .secure(true)
+            // .max_age(Duration::MAX) // do we want exp date?
+            // .domain(value) // TODO: set domain?
+            .build()
+    }
+}
+
+impl Default for UserState {
+    fn default() -> Self {
+        let selection = Vec::new();
+        UserState { selection }
+    }
+}
+
+impl<'a> TryFrom<Cookie<'a>> for UserState {
+    type Error = AppError;
+
+    fn try_from(cookie: Cookie<'a>) -> Result<Self, Self::Error> {
+        let cookie_base64 = cookie.value();
+
+        let cookie_json = STANDARD_NO_PAD.decode(cookie_base64).map_err(|err| {
+            error!("invalid base64 encoded cookie: {}", err);
+            StatusCode::BAD_REQUEST
```

Comment:
For the same reason stated below, this shouldn't be using status code at all IMO.

Reply:
I suspect you should be using  `Anyhow` instead?